docker/docker_client.go

```diff
@@ -388,31 +388,54 @@ func SearchRegistry(ctx context.Context, sys *types.SystemContext, registry, ima
 	}
 
 	logrus.Debugf("trying to talk to v2 search endpoint")
-	resp, err := client.makeRequest(ctx, "GET", "/v2/_catalog", nil, nil, v2Auth, nil)
-	if err != nil {
-		logrus.Debugf("error getting search results from v2 endpoint %q: %v", registry, err)
-	} else {
+	searchRes := []SearchResult{}
+	resultLimit := limit
+	path := "/v2/_catalog"
+	for resultLimit > 0 {
```

Comment:
The `resultLimit` name does not really say “the number of results we still want to get”.

Instead of worrying about naming it correctly, we don’t actually need that variable: `len(searchRes) < limit` computes the same thing in the same O(1) time, and we wouldn’t have to worry about keeping `searches` and `resultLimit` in sync.